docs/NamingConvention.md

```diff
@@ -1,37 +1,61 @@
 # Naming Convention [Proposal]
-
 ## Common naming criteria for all resources
 - Max length: `1-63` chars
 - Start with lower case character, can’t end in `-` or `_` or any special char
 > Preferred names have all lower case
 >> A preferred name can be regexed by `[a-z]([-a-z0-9]*[a-z0-9])?`
 
 ## Azure resource names
-Azure resources will have four parts:
-- `[ prefix ]-( project-short-form )-[ identifier ]-[ suffix ]-( environment-type )`
-  - Mandatory: `(project-short-form)`. Example: `ngsa`, `sp`
-  - Mandatory: `(environment-type)` - usage env. Possible val: `prod`, `test` and `dev`
-  - Optional: `[prefix]` or `[suffix]` - resource type suffix. Follows [Azure naming convention][1]
-    - Common resource prefix/suffix:
-        | Az Resource Name   | Prefix/Suffix |
-        |--------------------|----------|
-        | Key Vault          | kv-      |
-        | Function           | func-    |
-        | Virtual Machine    | vm-      |
-        | VM Scale Set       | vmss-    |
-        | CosmosDB           | cosmos-  |
-        | AKS Cluster        | aks-     |
-        | Container Registry | cr-      |
-  - Optional: `[identifier]` - refers to descriptive label
-  - Each part will be separated by a hyphen `-`
-  - Resource names will use either suffix or prefix…
-
-## Naming convention for Kubernetes resources 
-### Label format
-
-## Resourcese
+Azure resources consists of four parts: `( project-short-form )-[ identifier ]-( environment-type )-[ suffix ]`
```

Comment:
Does this include resource groups also? If not, I think it'll be helpful to add a snippet for it and providing some examples. Personally, the way helium (and helium-terraform) does it would be a good idea:
`<project>-rg-<summary label>`
Example: `ngsa-rg-app`

Reply:
A follow up to the resource group notation. If we use the same pattern for resource groups, we can end up with a following setup:
`ngsa-cosmos-pre-rg`

When these resource groups are sorted, the view will clump all of the resource groups for cosmos together. Rather, we should group the RGs by the environment, for example: `ngsa-pre-cosmos-rg`.

Reply:
definitely want ngsa-pre-*  It doesn't matter to me on the other two, but definitely want to sort by project-environment first IMO.

I like having the rg before the service because it makes deleting RGs from az easier, but I can live with it either way.